libs/net/traffic_generator.py

```
@@ -118,25 +129,18 @@ def _stop_process(vm: BaseVirtualMachine, cmd: str) -> None:
-            func=vm.console,
-            commands=[f"pgrep -fx '{cmd}'"],
+        vm.console(
+            commands=[f"pgrep -fAx '{cmd}'"],
             timeout=_DEFAULT_CMD_TIMEOUT_SEC,
-        ):
-            if sample:
-                return True
-    except TimeoutExpiredError as e:
-        LOGGER.warning(f"Process is not running on VM {vm.name}. Error: {str(e.last_exp)}")
+        )
+        return True
```

Ruff (0.14.1)

Consider moving this statement to an `else` block

(TRY300)